cli/cli.js

```diff
@@ -1,24 +1,39 @@
 #!/usr/bin/env node
-const { spawn } = require('child_process');
+const { spawn, spawnSync } = require('child_process');
 const path = require('path');
 const fs = require('fs');
 
 let electronBinary = process.env.ELECTRON_BINARY;
 
 if (!electronBinary) {
-  let pathStr =
-    process.platform === 'win32'
-      ? '../node_modules/electron/dist/electron.exe'
-      : '../node_modules/.bin/electron';
-
-  // recursively look for electron binary in node_modules folder
-  for (let i = 0; i < 7; i++) {
-    const maybeElectronBinary = path.resolve(__dirname, pathStr);
-    if (fs.existsSync(maybeElectronBinary)) {
-      electronBinary = maybeElectronBinary;
-      break;
+  // Check whether electron is installed globally and compare with the expected version
+  const electronHandleTemp = spawnSync('electron', ['--version']);
+
+  if (electronHandleTemp.stdout) {
+    if (!electronHandleTemp.stdout.toString().startsWith('v29.')) {
```

Comment:
any way this could reuse the requirement specification from the package.json?

Reply:
Yes I would like that more as well. I will spend some more time into whether/how that's possible

Reply:
Switched to injecting it at build time. Reading it at runtime would be non-trivial because the relative location of the package.json to where the binary is being run is unknown.